ImageInLib/ImageInLib/src/imageInterpolation.h

```diff
@@ -7,45 +7,99 @@ extern "C" {
+	* orientation : matrix for orientation/rotation IJK((1, 0, 0),(0,1,0),(0,0,1)), RAS((-1,0,0),(0,-1,0),(0,0,1))
+	* or LPS((0,0,1),(0,-1,0),(0,0,-1))
+	* For given point in real world cordinates system the function provide
+	* the corresponding point in image coordinates system
+	*/
+	Point3D getImageCoordFromRealCoord3D(Point3D src_point, Point3D real_origin, VoxelSpacing real_spacing, OrientationMatrix orientation);
+
+	/*
+	* Point3D getRealCoordFomImageCoord3D(Point3D src_point, Point3D real_origin, VoxelSpacing real_spacing, OrientationMatrix orientation)
+	* src_point : source 3D points in image coordinates system
+	* real_origin : origin in real world
+	* real_spacing : - distance between voxels in x and y direction
+	*                 - distance between slices
+	* orientation : matrix for orientation/rotation IJK((1, 0, 0),(0,1,0),(0,0,1)), RAS((-1,0,0),(0,-1,0),(0,0,1))
+	* or LPS((0,0,1),(0,-1,0),(0,0,-1))
+	* For given point in image cordinates system the function provide
+	* the corresponding point in real world coordinates system
+	*/
+	Point3D getRealCoordFomImageCoord3D(Point3D src_point, Point3D real_origin, VoxelSpacing real_spacing, OrientationMatrix orientation);
+
+	//=====================================================
+	//2D Functions
 
-	bool downSampling(dataType** originalImage, dataType** newImage, size_t length, size_t width, size_t height);
+	/*
+	* Point2D getImageCoordFromRealCoord2D(Point2D src_point, Point2D real_origin, PixelSpacing real_spacing, OrientationMatrix2D orientation)
+	* src_point : source 2D points in image coordinates system
+	* real_origin : origin in real world
+	* real_spacing : distance between voxels in x and y direction
+	* orientation : matrix for orientation
+	* For given point in real world cordinates system the function provide
+	* the corresponding point in image coordinates system
+	*/
+	Point2D getRealCoordFromImageCoord2D(Point2D src_point, Point2D real_origin, PixelSpacing real_spacing, OrientationMatrix2D orientation);
 
-	bool upSampling(dataType** originalImage, dataType** newImage, size_t length, size_t width, size_t height);
+	/*
+	* Point2D getImageCoordFromRealCoord2D(Point2D src_point, Point2D real_origin, PixelSpacing real_spacing, OrientationMatrix2D orientation)
+	* src_point : source 2D points in image coordinates system
+	* real_origin : origin in real world
+	* real_spacing : - distance between voxels in x and y direction
+	* orientation : matrix for orientation
+	* For given point in real world cordinates system the function provide
+	* the corresponding point in image coordinates system
+	*/
+	Point2D getImageCoordFromRealCoord2D(Point2D src_point, Point2D image_origin, PixelSpacing real_spacing, OrientationMatrix2D orientation);
 
-	//=========================
+	/*
+	* This function compute 2D euclidian distance between two given points
+	* point1 and point2 are given points
+	*/
+	dataType getDistance2D(Point2D point1, Point2D point2);
```

Comment:
we have similar function in common_functions.h)
getPoint2DDistance
Please use it and remove this one.

Reply:
yes I will use that one